pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -353,8 +354,10 @@ func NewNodeLifecycleController(podInformer coreinformers.PodInformer,
 				nc.nodeUpdateQueue.Add(node)
 				return nil
 			}),
-			UpdateFunc: nodeutil.CreateUpdateNodeHandler(func(_, newNode *v1.Node) error {
-				nc.nodeUpdateQueue.Add(newNode)
+			UpdateFunc: nodeutil.CreateUpdateNodeHandler(func(oldNode, newNode *v1.Node) error {
+				if !reflect.DeepEqual(oldNode.Status.Conditions, newNode.Status.Conditions) {
```

Comment:
I'm not sure if this is correct, as IIUC this means we would never change the taints if we failed to do it initially and the node conditions didn't change ever after that.

@wojtek-t

Reply:
So you're saying - if we fail all attempts of retrying, then we won't restart.

Yes - we should readd that node to the queue in that case.

Reply:
correct !

Reply:
> So you're saying - if we fail all attempts of retrying, then we won't restart.

Yes